components/api-resource-mgt/org.wso2.carbon.identity.api.resource.mgt/src/main/java/org/wso2/carbon/identity/api/resource/mgt/util/APIResourceManagementUtil.java

```diff
@@ -64,4 +67,24 @@ public static APIResourceMgtServerException handleServerException(
 
         return new APIResourceMgtServerException(error.getMessage(), description, error.getCode(), e);
     }
+
+    /**
+     * Restrict API resource creation for organizations.
+     * Check whether the tenant is an organization and return a client error if it is.
+     *
+     * @param tenantDomain Tenant domain.
+     * @throws APIResourceMgtException If an error occurs while checking whether the tenant is an organization.
+     */
+    public static void handleAPICreationInOrganization(String tenantDomain) throws APIResourceMgtException {
+
+        try {
+            if (Utils.isOrganization(tenantDomain)) {
```

Comment:
This will block the API resource creation in primary org as well. Isn't it?

Reply:
this only block API resource creation in child organizations (https://github.com/wso2/identity-organization-management-core/pull/93)